code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -106,69 +107,261 @@
+
+
+/obj/item/proc/capture_soul_process(var/mob/living/carbon/user, var/client/C, var/atom/movable/target, var/atom/movable/add_target = null)
+	//user is the guy using the soulstone
+	//C is the client of the guy we're soulstoning, so we don't lose track of him between the beginning and the end of the soulstoning.
+	//target is the source of the guy's soul (his body, or his head if decapitated)
+	//add_target is his body if he has been decapitated, for cosmetic purposes (and so it dusts)
+
+	if(!C)
+		return
+
+	var/mob/living/carbon/human/body = null
+
+	if(istype(target,/mob/living/carbon/human))
+		body = target
+	else if(istype(add_target,/mob/living/carbon/human))
+		body = add_target
+
+	var/true_name = "Unknown"
+
+	if(body)
+		true_name = body.real_name
+
+		for(var/obj/item/W in body)
+			body.drop_from_inventory(W)
+
+		body.dropBorers(1)
+
+		var/turf/T = get_turf(body)
+
+		body.invisibility = 101
+
+		var/datum/organ/external/head_organ = body.get_organ("head")
+		if(head_organ.status & ORGAN_DESTROYED)
+			new /obj/effect/decal/remains/human/noskull(T)
+			var/atom/movable/overlay/animation = new /atom/movable/overlay(T)
+			animation.icon_state = "blank"
+			animation.icon = 'icons/mob/mob.dmi'
+			animation.master = T
+			flick("dust-h2-nohead", animation)
+			animation.master = null
+			qdel(animation)
+		else
+			new /obj/effect/decal/remains/human(T)
+			if(body.lying)
+				var/atom/movable/overlay/animation = new /atom/movable/overlay(T)
+				animation.icon_state = "blank"
+				animation.icon = 'icons/mob/mob.dmi'
+				animation.master = T
+				flick("dust-h2", animation)
+				animation.master = null
+				qdel(animation)
+			else
+				var/atom/movable/overlay/animation = new /atom/movable/overlay(T)
+				animation.icon_state = "blank"
+				animation.icon = 'icons/mob/mob.dmi'
+				animation.master = T
+				flick("dust-h", animation)
+				animation.master = null
+				qdel(animation)
+
+		if(body.decapitated)
+			var/obj/item/weapon/organ/head/body_head = body.decapitated
+
+			body_head.invisibility = 101
+
+			new /obj/item/weapon/skull(get_turf(body.decapitated))
+	else
+		target.invisibility = 101
+
+		if(ismob(target))
+			var/mob/M = target
+			true_name = M.real_name
+			new /obj/effect/decal/cleanable/ash(get_turf(target))
+		else if(istype(target,/obj/item/weapon/organ/head))
+			var/obj/item/weapon/organ/head/H = target
+			var/mob/living/carbon/brain/BM = H.brainmob
+			true_name = BM.real_name
+			new /obj/item/weapon/skull(get_turf(target))
+
+	//Scary sound
+	playsound(get_turf(src), get_sfx("soulstone"), 50,1)
+
+	//Creating a shade inside the stone and putting the victim in control
+	var/mob/living/simple_animal/shade/S = new /mob/living/simple_animal/shade(get_turf(src))
+	S.loc = src //put shade in stone
+	S.status_flags |= GODMODE //So they won't die inside the stone somehow
+	S.canmove = 0//Can't move out of the soul stone
+	S.name = "Shade of [true_name]"
+	S.real_name = "Shade of [true_name]"
+	C.mob = S
```

Comment:
The standard way to do this is to set the ckey of the mob not change the mob of the client